marketplace-builder/src/builder.rs

```diff
@@ -501,8 +505,38 @@ mod test {
             .unwrap()
     }
 
-    #[async_std::test]
-    async fn test_marketplace_reserve_builder() {
+    /// Get the bundle associated with the received quorum proposal.
+    ///
+    /// # Returns
+    /// * `Some` if the bundle is fetched.
+    /// * `None` if the quorum proposal isn't received or the bundle fetching fails.
+    async fn get_bundle(
+        event: Event<SeqTypes>,
+        builder_client: Client<ServerError, MarketplaceVersion>,
+    ) -> Option<Bundle<SeqTypes>> {
+        if let EventType::QuorumProposal { proposal, .. } = event.event {
+            let parent_view_number = *proposal.data.view_number;
+            let parent_commitment = Leaf::from_quorum_proposal(&proposal.data).payload_commitment();
+            if let Ok(bundle) = builder_client
+                .get::<Bundle<SeqTypes>>(
+                    format!(
+                        "bundle_info/bundle/{parent_view_number}/{parent_commitment}/{}",
+                        parent_view_number + 1
+                    )
+                    .as_str(),
+                )
+                .send()
+                .await
+            {
+                return Some(bundle);
```

Comment:
I'm not sure whether it's correct to return `Some` without checking if `bundle.transactions` is empty. The public mempool tests now fail pretty consistently after I changed it to this way.

Reply:
I think one potential cause of this comes from assumping that the next event we get from the Network is a proposal.  I think we should be waiting for a `Transaction` event before we wait for this `Quorum Proposal` in order to see the transactions.

We might not actually want to wait for the `Quorum Proposal` either in these cases.  As receiving the `Transaction` event will cause any requests for subsequent `Quorum Proposal`s to return the transaction list until they are included.

Reply:
Looks like we can't use a `Transaction` event directly here, because we need the parent commitment to fetch the bundle. But I agree we shouldn't wait for the `QuorumProposal`. I'll think about what might be a good alternative.

Reply:
Oh, maybe I misunderstood your point. Do you mean we just do transaction comparison so there's no need to fetch the bundle?

Reply:
I should add that I tried adding this yesterday, but it didn't seem to result in a pass... instead I received errors about not being able to receive a bundle.

Reply:
As discussed I moved the quorum proposal handling earlier, so it corresponds to the parent view now, and the tests pass!